### PR TITLE
Added LocalNetworkConfig

### PR DESCRIPTION
**Pull request by mbjtrifork**

### CODE REVIEW

File: charts/flink-job/Chart.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.4.7
+version: 0.4.8
```
Comment (AndersBennedsgaard) [inline charts/flink-job/Chart.yaml:10]:
```suggestion
version: 0.5.0
```

File: charts/flink-job/ci/full-values.yaml [hunk 2]
```diff
@@ -224,3 +224,6 @@ ingress:
 
 image-automation:
   enabled: false
+
+local-network-configuration:
+  enabled: true
```
Comment (AndersBennedsgaard) [inline charts/flink-job/ci/full-values.yaml]: We should keep the camelcase formatting (please ignore `image-automation` :sweat_smile: )
```suggestion
localNetworkConfiguration:
  # -- Create a localNetworkConfiguration for allowing traffic between job- and task-manager in a deny-all environment
  enabled: true

image-automation:
  enabled: false

```

File: charts/flink-job/templates/local-network-config.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{{- if .Values.local-network-configuration.enabled -}}
+apiVersion: networking.tcs.trifork.com/v1alpha1
+kind: LocalNetworkConfig
+metadata:
+  name: local-network-config-{{ include "flink-job.fullname" . }}
```
Comment (AndersBennedsgaard) [inline charts/flink-job/templates/local-network-config.yaml:5]:
```suggestion
  name: {{ include "flink-job.fullname" . }}
```

File: charts/flink-job/templates/local-network-config.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+spec:
+  components:
+    {{ include "flink-job.fullname" . }}-job-component:
```
Comment (AndersBennedsgaard) [inline charts/flink-job/templates/local-network-config.yaml:8]:
```suggestion
    {{ include "flink-job.fullname" . }}-job:
```

File: charts/flink-job/templates/local-network-config.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      dependsOn:
+      - component: {{ include "flink-job.fullname" . }}-taskmanager-component
```
Comment (AndersBennedsgaard) [inline charts/flink-job/templates/local-network-config.yaml:10]:
```suggestion
      - component: {{ include "flink-job.fullname" . }}-taskmanager
```

File: charts/flink-job/templates/local-network-config.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        port: taskmanager-rpc
+        protocol: TCP
+      podSelector:
+        matchLabels:
+          app.kubernetes.io/instance: {{ .Release.Name }}
+          component: jobmanager
+    {{ include "flink-job.fullname" . }}-taskmanager-component:
```
Comment (AndersBennedsgaard) [inline charts/flink-job/templates/local-network-config.yaml:17]:
```suggestion
    {{ include "flink-job.fullname" . }}-taskmanager:
```

File: charts/flink-job/templates/local-network-config.yaml [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      dependsOn:
+      - component: {{ include "flink-job.fullname" . }}-job-component
```
Comment (AndersBennedsgaard) [inline charts/flink-job/templates/local-network-config.yaml:19]:
```suggestion
      - component: {{ include "flink-job.fullname" . }}-job
```

File: charts/flink-job/templates/local-network-config.yaml [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+        port: jobmanager-rpc
+        protocol: TCP
+      - component: {{ include "flink-job.fullname" . }}-job-component
```
Comment (AndersBennedsgaard) [inline charts/flink-job/templates/local-network-config.yaml:22]:
```suggestion
      - component: {{ include "flink-job.fullname" . }}-job
```

File: charts/flink-job/templates/local-network-config.yaml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+        port: blobserver
+        protocol: TCP
+      podSelector:
+        matchLabels:
+          app.kubernetes.io/instance: {{ .Release.Name }}
```
Comment (AndersBennedsgaard) [inline charts/flink-job/templates/local-network-config.yaml:27]:
```suggestion
          {{ include "flink-job.selectorLabels" . | nindent 10 }}
```

File: charts/flink-job/templates/local-network-config.yaml [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+spec:
+  components:
+    {{ include "flink-job.fullname" . }}-job-component:
+      dependsOn:
+      - component: {{ include "flink-job.fullname" . }}-taskmanager-component
+        port: taskmanager-rpc
+        protocol: TCP
+      podSelector:
+        matchLabels:
+          app.kubernetes.io/instance: {{ .Release.Name }}
```
Comment (AndersBennedsgaard) [inline charts/flink-job/templates/local-network-config.yaml:15]:
```suggestion
          {{ include "flink-job.selectorLabels" . | nindent 10 }}
```

File: charts/flink-job/values.yaml [hunk 11]
```diff
@@ -448,3 +448,7 @@ image-automation:
   # Image-automation is not possible when using image-sha as a tagging strategy.
   # Any other configuration given here, is passed to it
   enabled: false
+
+localNetworkConfiguration:
```
Comment (AndersBennedsgaard) [inline charts/flink-job/values.yaml:452]: Please move this above `image-automation`

**Review by mnrTrifork — APPROVED** [review]
LGTM

**Comment by mnrTrifork** [comment]
> LGTM

Glad @AndersBennedsgaard also reviewed this 😅

**Pull request merged**